packages/indexer-common/src/network.ts

```diff
@@ -244,17 +244,18 @@ export class Network {
     // --------------------------------------------------------------------------------
     const networkIdentifier = await networkProvider.getNetwork()
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let tapContracts: any
-    try {
-      tapContracts = await connectTapContracts(
-        wallet,
-        networkIdentifier.chainId,
-        specification.tapAddressBook,
-      )
-    } catch (err) {
-      logger.error(`Failed to connect to tap contract bindings:`, { err })
+    let tapContracts: any = undefined
+    if (tapSubgraph) {
+      try {
+        tapContracts = await connectTapContracts(
+          wallet,
+          networkIdentifier.chainId,
+          specification.tapAddressBook,
+        )
+      } catch (err) {
+        logger.error(`Failed to connect to tap contract bindings:`, { err })
```

Comment:
If there is a failure to connect do we want to only log and continue? Or should we exit out at that point?

Reply:
I guess you are right, in that point because if we do send a subgraph url we definitely desire TAP to work, and if contracts don't work what is the point

Reply:
Yup, I'd prefer to see a "fail fast" here